sklearn/mixture/tests/test_gaussian_mixture.py

```diff
@@ -1263,6 +1250,46 @@ def test_gaussian_mixture_setting_best_params():
         assert hasattr(gmm, attr)
 
 
+@pytest.mark.parametrize(
+    "init_params", ["random", "random_from_data", "k-means++", "kmeans"]
+)
+def test_check_non_unique_means_initialisation(init_params):
+    # Check that all initialisations provide unique starting means
+    rng = np.random.RandomState(0)
+    rand_data = RandomData(rng, scale=5)
+    n_components = rand_data.n_components
+    X = rand_data.X["full"]
+    default_means_init = {}
+
+    gmm = GaussianMixture(
+        n_components=n_components, random_state=0, init_params=init_params, max_iter=1
+    )
+    gmm.fit(X)
+    default_means_init[init_params] = gmm.means_
+    for i_mean, j_mean in itertools.combinations(default_means_init.values(), r=2):
+        assert np.any(np.not_equal(i_mean, j_mean))
```

Comment:
Following my comment above, we could instead check that
- the fitted means have the expected shape
- the fitted means are "within" the data, i.e. between xmin and xmax for each feature
- it managed to converge, i.e. ``converged_`` is True
```suggestion

assert gmm.means_.shape == (n_components, n_features)
assert np.all(X.min(axis=0) <= gmm.means_)
assert np.all(gmm.means_ <= X.max(axis=0))
assert gmm.converged_
```